relay-server/src/utils/rate_limits.rs

```diff
@@ -1907,4 +2148,510 @@ mod tests {
 
         assert_eq!(get_outcomes(enforcement), vec![(DataCategory::LogByte, 20)]);
     }
+
+    #[tokio::test]
+    async fn test_enforce_standalone_span_attachment() {
+        let test_cases = vec![
+            (
+                "span_limit",
+                vec![DataCategory::Span],
+                true,
+                vec![(DataCategory::Span, 0)],
+                vec![
+                    (DataCategory::Attachment, 7),
+                    (DataCategory::AttachmentItem, 1),
+                ],
```

Comment:
```suggestion
                "span_limit",
               &[DataCategory::Span],
                true,
                &[(DataCategory::Span, 0)],
                &[
                    (DataCategory::Attachment, 7),
                    (DataCategory::AttachmentItem, 1),
                ],
```

No need for the vectors I think